packages/connect-wallet-modal/src/components/WalletsModal/components/ConnectWalletModal/sortWalletsList.ts

```diff
@@ -7,41 +7,38 @@ type GetWalletsListArgs = {
   walletsPinned: WalletsModalProps['walletsPinned'];
 };
 
-export function sortWalletsList({
+export async function sortWalletsList({
   walletDataList,
   walletsShown,
   walletsPinned,
 }: GetWalletsListArgs) {
-  const filteredWalletData = walletsShown.reduce(
-    (walletsList, walletId) => {
-      const walletData = walletDataList.find((w) => w.walletId === walletId);
-
-      if (!walletData) return walletsList;
-
-      const { detector, autoConnectOnly } = walletData;
-
-      // Filtering wallets marked as hidden and auto connect only
-      if (autoConnectOnly) return walletsList;
-
-      if (walletsPinned.includes(walletId)) {
-        // Put the pinned wallets on the first place, above all another
-        walletsList.pinned.push(walletData);
-      } else if (detector?.()) {
-        // If condition is true (usually means that a wallet is detected),
-        // move it to the first place in the wallets list, so a user can see it right away
-        walletsList.detected.push(walletData);
-      } else {
-        walletsList.default.push(walletData);
-      }
-
-      return walletsList;
-    },
-    {
-      pinned: [] as WalletAdapterData[],
-      detected: [] as WalletAdapterData[],
-      default: [] as WalletAdapterData[],
-    },
-  );
+  const filteredWalletData = {
```

Comment:
Suggest this code style:

type FilteredWalletData = {
pinned: WalletAdapterData[],
detected: WalletAdapterData[],
default: WalletAdapterData[]
}

const filteredWalletData: FilteredWalletData = {...}